File: QytetetJava/src/modeloqytetet/PruebaQytetet.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

import java.util.ArrayList;


/**
 *
 * @author juliu
 */
public class PruebaQytetet {
    //private static ArrayList<Sorpresa> mazo = new ArrayList();
    //private static Tablero tablero = new Tablero();
    /**
     * @param args the command line arguments
     */

    
   /* private static ArrayList mayorQueCero(){
        
        ArrayList<Sorpresa> nuevo_mazo = new ArrayList();
        for (Sorpresa imazo : mazo) {
            if(imazo.getValor() > 0){
                nuevo_mazo.add(imazo);
            }
        }
        return nuevo_mazo;        
    }
    
    private static ArrayList tipoIrCasilla(){
        
        ArrayList<Sorpresa> nuevo_mazo = new ArrayList();
        for (Sorpresa imazo : mazo) {
            if(imazo.getTipo() == TipoSorpresa.IRACASILLA){
                nuevo_mazo.add(imazo);
            }
        }
        return nuevo_mazo;        
    }
    
    private static ArrayList compararCasilla(TipoSorpresa tip){
        
        ArrayList<Sorpresa> nuevo_mazo = new ArrayList();
        for (Sorpresa imazo : mazo) {
            if(imazo.getTipo() == tip){
                nuevo_mazo.add(imazo);
            }
        }
        return nuevo_mazo;        
    }
        
    void imprimirPrivadas(){
        ArrayList<Sorpresa> mazito = new ArrayList();
        
        mazito = mayorQueCero();
        for (Sorpresa mazo1 : mazito) {
            System.out.println(mazo1.toString());
        } 
        
        mazito = tipoIrCasilla();
        for (Sorpresa mazo1 : mazito) {
            System.out.println(mazo1.toString());
        } 
        
    
        
    }  
    */
    public static void main(String[] args) {
        
        // TODO code application logic here
        Qytetet prueba = Qytetet.getInstance();
        ArrayList<String> nombres = new ArrayList();
        
        nombres.add("jugador1");
        nombres.add("jugador2");
        
        prueba.inicializarJugadores(nombres);
        
        System.out.println(prueba.toString());
   //   prueba.inicializarCartasSorpresa();
   //   prueba.inicializarTablero();
        ArrayList<Sorpresa> mazito = new ArrayList();
        
        /*mazito = mayorQueCero();
        for (Sorpresa mazo1 : mazito) {
            System.out.println(mazo1.toString());
        }
        
        mazito = tipoIrCasilla();
        for (Sorpresa mazo1 : mazito) {
            System.out.println(mazo1.toString());
        } */
        
        
        //Jugador jugador2 = new Jugador("longuerson");
        
         
        
   /*     
        System.out.println(tablero.toString());
        System.out.println(tablero.getCarcel().getNumeroCasilla());
     */   
    }
       
} 
        

File: QytetetRuby/lib/prueba_qytetet.rb
#encoding: utf-8
#To change this license header, choose License Headers in Project Properties.
#To change this template file, choose Tools | Templates
#and open the template in the editor.

module ModeloQytetet
  

require_relative "sorpresa"
require_relative "tipo_sorpresa"
require_relative "tablero"
require_relative "titulo_propiedad"
require_relative "tipo_casilla"
require_relative "casilla"
require_relative "qytetet"

  
  class PruebaQytetet
   
    
 
 #   @@mazo = Array.new
  
  #  def self.inicializar_sorpresas
  #    @@mazo<< Sorpresa.new("te hemos pillado haciendo cosas que no debias, ¡a la carcel!", 9, TipoSorpresa::IRACASILLA)
  #    @@mazo<< Sorpresa.new("Alguien ha pagado tu fianza, salesde la carcel.", 0, TipoSorpresa::SALIRCARCEL)
  #    @@mazo<< Sorpresa.new("Mala suerte amigo, has molestado a un pez gordo. ¡A pagar!", 200, TipoSorpresa::PAGARCOBRAR)
  #    @@mazo<< Sorpresa.new("Te has levantado con la pierna derecha, has ganado un premio", 100, TipoSorpresa::PAGARCOBRAR)
  #    @@mazo<< Sorpresa.new("Te ha abducido un ovni y te ha dejado en la 10ª avenida", 10, TipoSorpresa::IRACASILLA)
  #    @@mazo<< Sorpresa.new("Has cogido un taxi y te lleva al 23", 23, TipoSorpresa::IRACASILLA)
  #    @@mazo<< Sorpresa.new("Hacienda te ha pillado te toca pagar una pasta por tus propiedades", 50, TipoSorpresa::PORCASAHOTEL)
  #    @@mazo<< Sorpresa.new("Hacienda se ha equivocado contigo recibes una \"gratificación\" ", 120, TipoSorpresa::PORCASAHOTEL)
  #    @@mazo<< Sorpresa.new("Tus compañeros estan generosos hoy", 100, TipoSorpresa::PORJUGADOR)
  #    @@mazo<< Sorpresa.new("Te has portado mal con tus colegas asi que les indemnizas", 120, TipoSorpresa::PORJUGADOR)
  #  end
  
  #  def self.mayorQueCero
    
  #    nuevo_mazo = Array.new  
  #    for nmazo in @@mazo    
  #      if nmazo.valor > 0 
  #        nuevo_mazo << nmazo 
  #      end   
  #    end   
  #  end
    
    
  
  #  def self.tipoIrCasilla
  #  
  #    nuevo_mazo = Array.new
  #    for nmazo in @@mazo
  #      if nmazo.tipo == TipoSorpresa::IRACASILLA
  #        nuevo_mazo << nmazo
  #      end
  #    end
  #  end
  #
  #  def self.compararCasilla(tipo_sorpresa)
  #    nuevo_mazo = Array.new
  #    for nmazo in @@mazo
  #      if nmazo.tipo == tipo_sorpresa
  #        nuevo_mazo << nmazo
  #      end
  #    end
  #  end
  #
  def self.main
      
     # inicializar_sorpresas()
      #puts mayorQueCero()
     # puts tipoIrCasilla
     # puts compararCasilla(TipoSorpresa::PORJUGADOR)
     # tablero = Tablero.new
     # puts tablero.to_s
      
      prueba = Qytetet.instance
      nombres = Array.new
      nombres << "jugador 1"
      nombres << "jugador 2"
      
      
      prueba.inicializar_jugadores(nombres)
      #prueba.inicializar_cartas_sorpresa
      #prueba.inicializar_tablero
      puts prueba.to_s
      
    end
  
  end
  PruebaQytetet.main
end